Console/Command/NodesValidatorCommand.php

```diff
@@ -0,0 +1,203 @@
+        $this->validator = $validator;
+        $this->validationAggregateError = $validationAggregateError;
+        $this->state = $state;
+
+        parent::__construct($name);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $description = 'It will loop all menu nodes and return the IDs and description of invalid menu nodes.'
+            . PHP_EOL
+            . ' Additionally, it will ask to remove the detected invalid node if needed.';
+
+        $this->setName('snowdog-menu:validate-nodes')
+            ->setDescription($description);
+
+        parent::configure();
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->setAreaCode();
+        $invalidNodeIds = [];
+
+        foreach ($this->getAllMenus() as $menu) {
+            $output->writeln(PHP_EOL . '<info>... Validating nodes for menu with ID: ' . $menu->getMenuId(). ' </info>');
+            $nodes = $this->nodeRepository->getByMenu($menu->getMenuId());
+            $menuInvalidNodes = $this->getInvalidNodes($nodes);
+
+            if (empty($menuInvalidNodes)) {
+                $output->writeln('<info>No invalid nodes encountered</info>');
+                continue;
+            }
+
+            $output->writeln(
+                '<error>Invalid nodes encountered for menu ID: ' . $menu->getMenuId() . '</error>'
+            );
+
+            foreach ($menuInvalidNodes as $errorMessages) {
+                foreach ($errorMessages as $errorMessage) {
+                    $output->writeln("<error>- $errorMessage</error>");
+                }
+            }
+
+            $invalidNodeIds = array_merge($invalidNodeIds, array_keys($menuInvalidNodes));
+        }
+
+        if (empty($invalidNodeIds) || !$input->isInteractive()) {
+            return;
+        }
+
+        $helper = $this->getHelper('question');
+
+        if (!$helper->ask($input, $output, $this->getConfirmationQuestion())) {
+            $output->writeln(PHP_EOL . '<info>Invalid nodes were NOT removed</info>');
+            return;
+        }
+
+        $this->displayResults(
+            $this->deleteNodesByIds($invalidNodeIds),
+            $output
+        );
+
+        return 0;
+    }
+
+    private function setAreaCode(): void
+    {
+        try {
+            $this->state->getAreaCode();
+        } catch (LocalizedException $e) {
+            $this->state->setAreaCode(Area::AREA_ADMINHTML);
+        }
+    }
+
+    /**
+     * @return MenuInterface[]
+     */
+    private function getAllMenus(): array
+    {
+        return $this->menuRepository
+            ->getList($this->searchCriteriaBuilder->create())
+            ->getItems();
+    }
+
+    private function getInvalidNodes(array $nodes): array
+    {
+        $invalidNodes = [];
+
+        /** @var NodeInterface $node */
+        foreach ($nodes as $node) {
+            $this->validator->validate([$node->getNodeId() => $node->getData()]);
```

Comment:
Import validator has `ZERO_BASED_ARRAY_INDEXING_KEY_ADDENED` which makes the later output contain wrong node ids, using `disableNodeIdAddend` should help avoid that

It still deletes nodes using the proper node ids, fixing this would address the wrong node ids under `Invalid nodes encountered for menu ID`